pkg/controller/certificates/trigger/policies/gatherer.go

```diff
@@ -38,6 +39,29 @@ type Gatherer struct {
 	SecretLister             corelisters.SecretLister
 }
 
+// DataForCertificate returns the secret as well as the "current"
+// certificate request associated with the given certificate. It also
+// returns the given certificate as-is.
+//
+// The "current" certificate request designates certificate request that
+// led to the current revision of the certificate. The "current"
+// certificate request is by definition in a ready state, and can be seen
+// as the source of information of the current certificate.
+//
+// This "current" certificate request is not to be confused with the "next"
+// certificate request that you might get by listing the CRs for the
+// certificate's revision+1; these "next" CRs might not be ready yet.
+//
+// We need the "current" certificate request because this CR contains the
+// "source of truth" of the current certificate, and getting the "current"
+// CR allows is to check whether the current certificate still matches the
+// already-issued certificate request.
+//
+// This function does not return any apierrors.NewNotFound errors for
+// either the secret or the certificate request; instead, the secret and
+// certificate request are left nil. An error is returned when two
+// certificate requests are found for the couple (certificate's revision,
+// certificate's uid).
 func (g *Gatherer) DataForCertificate(ctx context.Context, crt *cmapi.Certificate) (Input, error) {
```

Comment:
Does it make sense? 😅

cc @munnerz

Reply:
Yes it does 💯 though I wonder if we can put a similar description onto the actual fields on the `Input` struct?

Reply:
Done!